### PR TITLE
Feat: add Pull Requests template

### LINKED ISSUE #33: Add Issue and PR templates
See Github documentation:
- [x] [ issue templates](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository)
- [ ] [Pull request templates](https://docs.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)

Exact format needs to be discussed.
[terraform-provider-oci issue templates](https://github.com/terraform-providers/terraform-provider-oci/tree/master/.github/ISSUE_TEMPLATE) should be a good point to start with.

### PR DESCRIPTION
**Pull request by kral2**
the PR template is customized for Terraform workflow and adapted to our current code organization, calling out for specific actions before submitting a PR in the form of checklists:

- proposed change description
- type of change checkbox
- testing checklist
- documentation update checklist

The content is generic enough to be adopted by each Terraform modules we maintain.

Later, we could enforce a PR Check using GitHub Actions to ensure every checklist are cleared before allowing code 
merge.

Fix #33

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Proposed change
+
+<!--- 
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+--->
+
+Fixed issues: <!--insert issue reference in this form: #XX #XX-->
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:7]: This will not auto-link those issues for closure. You need the format "Fix/fxes/fixed" or similar.

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Type of change
+
+<!--- Please delete options that are not relevant for your change --->
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:11]: If it's a checkbox list, why should I delete options too? Either I should check a box or delete.

File: .github/PULL_REQUEST_TEMPLATE [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+### Manual testing
+
+If no automated testing is run, please ensure that al least the three steps below are passing without any error.
+
+- [ ] Running `terraform apply` on each examples of this module provision the intended resources without any error
+- [ ] Modifying module Input Variables after initial provisioning act as intended: updatable properties gets updated without a resource recreate
+- [ ] Running `terraform destroy` on each examples of this module destroy all and only the resources created by this module
+
+## Checklist before submitting your PR
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:35]: Folks need to have signed the OCA as well: https://oca.opensource.oracle.com.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE (hunk 2 above)
Comment (kral2) [reply .github/PULL_REQUEST_TEMPLATE:11]: Correct, that may not be the right format for this section.

I am thinking about an action that checks if all tasks are complete (e.g: tasks 13/13). The correct format would be a radio button to permit only 1 option at a time, but I don't think it is supported.

Comment (kral2) [reply .github/PULL_REQUEST_TEMPLATE:11]: removing checkbox list and keep the "please delete options ..." notice.

File: .github/PULL_REQUEST_TEMPLATE [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Proposed change
+
+<!--- 
+Please include a summary of the change and which issue is fixed in the form of fix #XX for auto-linking.
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:4]: I would map a closing keyword to the type of change, i.e. `[Fixes|Resolves] #000` (bug fixes) and `Closes #000`  (new feature)

File: .github/PULL_REQUEST_TEMPLATE [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Proposed change
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:1]: **For your consideration; totally optional**

You may want to consider adopting a more formal commit taxonomy like Conventional Commits: https://www.conventionalcommits.org/en/v1.0.0/#summary

If you get this right, you can progress to Semantic Releases: https://semantic-release.gitbook.io/semantic-release/

File: .github/PULL_REQUEST_TEMPLATE [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+<!--- 
+Please include a summary of the change and which issue is fixed in the form of fix #XX for auto-linking.
+
+- Include relevant motivation and context
+- List any dependencies that are required for this change.
+--->
+
+## Type of change
+
+<!--- Please delete options that are not relevant for your change --->
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:12]: **Optional! This is how Conventional Commits handles this:**

Instead of deleting options, consider requiring that the subject of the PR and/or commit should denote the type of change, i.e. either `fix: .....` or `feat: ....` and breaking changes should either be:
- marked by adding two blank lines after the commit message then adding `BREAKING CHANGE:` and a description of what is breaking
- marked by adding a `!` after the type of change in the commit subject, i.e. `fix! blah blah` or `feat! blah blah` with a description of what is broken in the commit message.

The benefit of this is that you can scan `git log` to determine if you need to bump the patch (`fix`), minor (`feat`) or major (`breaking change` or `!`) version.

File: .github/PULL_REQUEST_TEMPLATE [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+- Bug fix (non-breaking change which fixes an issue)
+- New feature (non-breaking change which adds functionality)
+- Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## How has this change been tested?
+
+### Automated testing
+
+If your are running automated testing for this module, we would love to hear from you, and potentially integrate it to the module standard workflow.
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:22]: `s/your/you're`

File: .github/PULL_REQUEST_TEMPLATE [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+<!---
+Please describe the tests that you ran to verify your changes and provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
+--->
+
+### Manual testing
+
+If no automated testing is run, please ensure that at least the three steps below are passing without any error.
+
+- [ ] Running `terraform apply` on each examples of this module provision the intended resources without any error
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:32]: I would rephrase this as "Running `terraform apply` on each example provided with this module provisions the intended resource(s) without any errors."

File: .github/PULL_REQUEST_TEMPLATE [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] Modifying module Input Variables after initial provisioning act as intended: updatable properties gets updated without a resource recreate
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:33]: "Modifying the module's _input variables_ after initial provisioning behaves as intended, i.e. any updateable properties are ameneded without recreation of the resource(s)."

File: .github/PULL_REQUEST_TEMPLATE [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] Running `terraform destroy` on each examples of this module destroy all and only the resources created by this module
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:34]: Running `terraform destroy` on each example provided with this module destroys all the resources created by this module and only the resources created by this module."

File: .github/PULL_REQUEST_TEMPLATE [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+## Checklist before submitting your PR
+
+- [ ] My code follows [the style guidelines of this project](../tree/main/docs/codingconventions.adoc)
+- [ ] If contribution provision new resources, I updated the README introduction section (both adoc and md versions)
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:39]: I would break this down into individual tasks:

- [ ] This contribution provisions new resources
  - [ ] I have updated the <link to: ADOC version> of the README
  - [ ] I have updated the <link to: MD version> of the README

File: .github/PULL_REQUEST_TEMPLATE [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] If contribution adds any new variables, I updated [docs/terraformoptions.adoc](../tree/main/docs/terraformoptions.adoc)
+- [ ] I added an entry that explains the fix or enhancement to the list of unreleased items in CHANGELOG
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:41]: - [ ] I have updated the chanelog to include an entry for this contribution

File: .github/PULL_REQUEST_TEMPLATE [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] I updated all examples, including their README files and sample code blocks in it
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:42]: - [ ] I have updated all provided examples, including each README file and all applicable code blocks

File: .github/PULL_REQUEST_TEMPLATE [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] My changes generate no new warnings
+- [ ] Any dependent changes have been merged and published in downstream modules
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:44]: I'm not sure I'm following this one: aren't depdendent changes made in upstream modules?

File: .github/PULL_REQUEST_TEMPLATE [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+- [ ] Running `terraform destroy` on each examples of this module destroy all and only the resources created by this module
+
+## Checklist before submitting your PR
+
+- [ ] My code follows [the style guidelines of this project](../tree/main/docs/codingconventions.adoc)
+- [ ] If contribution provision new resources, I updated the README introduction section (both adoc and md versions)
+- [ ] If contribution adds any new variables, I updated [docs/terraformoptions.adoc](../tree/main/docs/terraformoptions.adoc)
+- [ ] I added an entry that explains the fix or enhancement to the list of unreleased items in CHANGELOG
+- [ ] I updated all examples, including their README files and sample code blocks in it
+- [ ] My changes generate no new warnings
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:43]: This contribution generates no new warnings

File: .github/PULL_REQUEST_TEMPLATE (hunk 5 above)
Comment (Djelibeybi) [reply .github/PULL_REQUEST_TEMPLATE:1]: **For all the rest of my comments**, feel free to replace "this contribution" with "my changes" or "these changes" or "this submission" or "this change".

File: .github/PULL_REQUEST_TEMPLATE [hunk 16]
```diff
@@ -0,0 +1,46 @@
+# Proposed change
+
+<!--- 
+Please include a summary of the change and which issue is fixed in the form of fix #XX for auto-linking.
+
+- Include relevant motivation and context
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:6]: Wouldn't the relevant motivation and context be in the linked issue? Not sure of the value of repeating it in the PR as well.

File: .github/PULL_REQUEST_TEMPLATE [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+- List any dependencies that are required for this change.
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:7]: "List any additional required dependencies"

File: .github/PULL_REQUEST_TEMPLATE [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+
+- [ ] Running `terraform apply` on each examples of this module provision the intended resources without any error
+- [ ] Modifying module Input Variables after initial provisioning act as intended: updatable properties gets updated without a resource recreate
+- [ ] Running `terraform destroy` on each examples of this module destroy all and only the resources created by this module
+
+## Checklist before submitting your PR
+
+- [ ] My code follows [the style guidelines of this project](../tree/main/docs/codingconventions.adoc)
+- [ ] If contribution provision new resources, I updated the README introduction section (both adoc and md versions)
+- [ ] If contribution adds any new variables, I updated [docs/terraformoptions.adoc](../tree/main/docs/terraformoptions.adoc)
```
Comment (Djelibeybi) [inline .github/PULL_REQUEST_TEMPLATE:40]: As above:

- [ ] This contribution adds new variables: `[List the new variable names]`
  - [ ] I have updated `docs/terraformoptions.adoc` to include each variable.

**Review by Djelibeybi — COMMENTED** [review]
Here is my feedback. I'm marking it as a comment so this PR can be merged without changing any of it.

File: .github/PULL_REQUEST_TEMPLATE (hunk 5 above)
Comment (kral2) [reply .github/PULL_REQUEST_TEMPLATE:1]: That would be great. I have seen a lot of initiatives in this space, but Angular Commit Message Conventions (on which semantic-release is based) stands out by being clean and straight forward 👍

File: .github/PULL_REQUEST_TEMPLATE (hunk 14 above)
Comment (kral2) [reply .github/PULL_REQUEST_TEMPLATE:44]: correct. adjusting.

**Review by Djelibeybi — APPROVED** [review]
LGTM
